learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
+```
+
+There are three things you should pay attention to when using an SDK to generate a tenant token. In the above example, they are stored in `searchRules`, `apiKey`, and `expiryDate`.
+
+`searchRules` must be an object containing at least one search rule. A search rule must specify either an index name or a `*` wildcard. Search rules must also declare at least one filter. In this example, the filter makes it so a user only has access to their own records.
+
+`apiKey` must be a valid Meilisearch API key. A tenant token will have access to the same indexes as the API key you use when generating it. If no API key is provided, the SDK might be able to infer it automatically.
+
+`expiryDate` is optional when using an SDK. If used, this parameter must be a UNIX timestamp. Tokens becomes invalid after their `expiryDate`.
+
+You can read more about each of element of the tenant token payload in [this guide's final section](#tenant-token-payload-reference).
+
+### Using a tenant token with an SDK
+
+The resulting token, stored on the `token` variable, can then be sent back to the front-end and used to make queries that will only return results whose `user_id` attribute equals the current user's ID:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_sdk_search_1">
+const frontEndClient = new Meilisearch({
+  apiKey: token
+});
+
+const medicalRecordsIndex = frontEndClient.index('patient_medical_records');
+const search = medicalRecordsIndex.search('blood test');
+```
+
+Applications may use tenant tokens and API keys interchangeably when searching. For example, the same application might use a default search API key for queries on public indexes and a tenant token for logged-in users searching on private data.
+
+## Generating tenant tokens without a Meilisearch SDK
+
+Though [Meilisearch recommends using an official SDK](#generating-tenant-tokens-with-an-sdk), this is not a requirement. Since tenant tokens follow the [JWT standard](https://jwt.io), you can use a number of [compatible third-party helper libraries](https://jwt.io/libraries). You may also skip all assistance and generate a token from scratch.
+
+The full process requires you to [create a token header](#step-1-creating-a-token-header), [prepare a data payload](#step-2-preparing-the-token-payload) with at least one set of search rules, and then [sign the token](#step-3-signing-a-token) with an API key.
+
+If you are already familiar with the creation of JWTs and only want to know about the specific requirements of a tenant token payload, take a look at this guide's [token payload reference](#tenant-token-payload-reference).
+
+::: note
+The following examples use JavaScript for the sake of simplicity, but Meilisearch should be compatible with the majority of modern stacks and languages.
+:::
+
+### Step 1: Creating a token header
```

Comment:
I know this might be late into the party but I would very much suggest you make an example using the [jsonwebtoken](https://www.npmjs.com/package/jsonwebtoken) library. 
Apart from the SDK's it's very very rare to create a JWT from scratch. I could barely find documentation online. 
Most of guides, [google's guide as well](https://cloud.google.com/iot/docs/how-tos/credentials/jwts), makes the example using that library!